conbench/tests/api/test_results.py

```diff
@@ -344,6 +344,17 @@ def test_benchmark_list_filter_by_run_id(self, client):
         response = client.get("/api/benchmarks/?run_id=200")
         self.assert_200_ok(response, [_expected_entity(benchmark_result)])
 
+    def test_benchmark_list_filter_by_run_reason(self, client):
+        self.authenticate(client)
+        thisresult = _fixtures.benchmark_result(run_id="100", reason="rolf")
+        resp = client.get("/api/benchmark-results/?run_reason=rolf")
+        assert resp.status_code == 200, resp.text
+        self.assert_200_ok(resp, [_expected_entity(thisresult)])
+
+        resp = client.get("/api/benchmark-results/?run_reason=foo")
+        assert resp.status_code == 200, resp.text
+        assert resp.json == []
```

Comment:
negative test: when asking for another run reason, the result is an empty list.

I think this confirms that the filter works.